catalog/resourcemanagement/resourcemanagement-usage-ui/src/main/webapp/js/model/DataUsage.js

```diff
@@ -0,0 +1,173 @@
+                        notify : (usageRemaining === "0 MB" || usageRemaining === "0 GB")};
+                    dataModel.push(object);
+                }));
+               return dataModel;
+            },
+            submitUsageData : function(data) {
+                var that = this;
+                var url = "/jolokia/exec/org.codice.ddf.resourcemanagement.usage.service.DataUsage:service=datausage/updateUserDataLimit/" + JSON.stringify(data);
+                $.ajax({
+                    url: url,
+                    dataType: 'json',
+                    success : function() {
+                        that.pollUntilUpdated();
+                    }
+                });
+            },
+            pollUntilUpdated: function() {
+                var that = this;
+                var updatedModel = false;
+                this.set({'saving' : true });
+
+               (function poll() {
+                    var currentModel = that.get("users") ;
+
+                    if(!updatedModel) {
+                        setTimeout(function() {
+                            $.ajax({
+                                url : "/jolokia/exec/org.codice.ddf.resourcemanagement.usage.service.DataUsage:service=datausage/userMap/",
+                                success: function(data) {
+                                    var receivedData = that.parseDataModel(data.value);
+                                    var current = JSON.stringify(currentModel);
+                                    var newData = JSON.stringify(receivedData);
+
+                                    if(newData !== current) {
+                                        that.set({'users' : receivedData});
+                                        updatedModel =  true;
+                                        that.set({'saving' : false });
+                                    }
+                                },
+                                dataType: "json",
+                                complete : poll
+                            });
+                        }, 500);
+                    }
+                })();
+            },
+            isLimitChanged : function(user, value, dataSize) {
+                var model = this.get('users');
+                var dataInBytes = value;
+
+                var isLimitChanged = false;
+                if(dataSize === "GB") {
+                    dataInBytes = dataInBytes * (1000 * 1000 * 1000);
+                } else {
+                    dataInBytes = dataInBytes * (1000 * 1000);
+                }
+
+                $.each(model, function(index, object) {
+                    if(object.user === user && dataInBytes !== object.usageLimit) {
+                        isLimitChanged = true;
+                    }
+
+                });
+                return isLimitChanged;
+            },
+            constructUsageRemainingString : function(dataLimit, dataUsage) {
+                    var gb = (1000 * 1000 * 1000);
```

Comment:
I do see `1000 * 1000 * 1000` repeated several times in this file. Repeated code makes my brain itch ;-)